chain-signatures/node/src/protocol/signature.rs

```diff
@@ -811,4 +812,69 @@ impl SignatureManager {
             .and_modify(|e| *e = Instant::now());
         matches!(entry, Entry::Occupied(_))
     }
+
+    pub fn execute(
+        state: &RunningState,
+        stable: &Participants,
+        me: Participant,
+        protocol_cfg: &ProtocolConfig,
+        ctx: &impl super::cryptography::CryptographicCtx,
+    ) -> tokio::task::JoinHandle<()> {
+        let threshold = state.threshold;
+        let my_account_id = state.triple_manager.my_account_id.clone();
+        let presignature_manager = state.presignature_manager.clone();
+        let signature_manager = state.signature_manager.clone();
+        let messages = state.messages.clone();
+        let stable = stable.clone();
+        let protocol_cfg = protocol_cfg.clone();
+        let sign_queue = state.sign_queue.clone();
+        let rpc_client = ctx.rpc_client().clone();
+        let signer = ctx.signer().clone();
+        let mpc_contract_id = ctx.mpc_contract_id().clone();
+
+        // NOTE: signatures should only use stable and not active participants. The difference here is that
+        // stable participants utilizes more than the online status of a node, such as whether or not their
+        // block height is up to date, such that they too can process signature requests. If they cannot
+        // then they are considered unstable and should not be a part of signature generation this round.
+
+        tokio::task::spawn(tokio::task::unconstrained(async move {
+            let mut sign_queue = sign_queue.write().await;
```

Comment:
Again, we acquire a write lock for shared state, hence only one task at the time will get passed this line.

Reply:
`sign_queue` should actually just be turned into async channel here because it's only used by the indexer and the `SignatureManager` thus eliminating the need for it to be a shared state

Reply:
I was also thinking about moving signature requests to Redis. This is part of preparations for horizontal scaling, where multiple servers will need to access the same pool of signature requests. Also, it is a step forward to a design, where we have one or more separate indexers that are adding requests to the queue.